tests/core/v5_1/alexandria/test_alexandria_network.py

```diff
@@ -150,3 +151,101 @@ async def test_network_explore(tester, alice):
 
         # Ensure that one of the three closest node ids was in the returned node ids
         assert best_node_ids_by_distance.intersection(found_node_ids)
+
+
+@pytest.mark.trio
+async def test_alexandria_network_find_content_api(
+    alice, bob, alice_alexandria_network, bob_alexandria_client
+):
+    response_enr = ENRFactory()
+    response_content = b'response-value'
+
+    async with bob_alexandria_client.subscribe(FindContentMessage) as subscription:
+        async with trio.open_nursery() as nursery:
+
+            async def _respond():
+                request = await subscription.receive()
+
+                await bob_alexandria_client.send_found_content(
+                    request.sender_node_id,
+                    request.sender_endpoint,
+                    enrs=(response_enr,),
+                    content=None,
+                    request_id=request.request_id,
+                )
+
+                request = await subscription.receive()
+
+                await bob_alexandria_client.send_found_content(
+                    request.sender_node_id,
+                    request.sender_endpoint,
+                    enrs=None,
+                    content=response_content,
+                    request_id=request.request_id,
+                )
+
+            nursery.start_soon(_respond)
+
+            with trio.fail_after(2):
+                response_a = await alice_alexandria_network.find_content(
+                    bob.node_id,
+                    content_key=b'test-key',
```

Comment:
> Also, we have to decide what to do when someone sends us bad data. Do we simply ignore it? Drop the peer? Stop asking them for anything over the next n seconds?

For what we're planning to use this codebase for, I don't think this will matter that much since we should be operating in a non-adversarial environment for a while, and once we're not, the python client isn't likely to see continued development.

Reply:
Right, more thinking about the spec & recommendation for other clients. Agree it's not a blocker for this PR :+1: